### PR TITLE
PR: Show symbols for namespace packages in the Outline

### LINKED ISSUE #16352: Outline explorer seeing impported symbols from relative import with tuple syntax
### What steps will reproduce the problem?

I have a django project that does some imports as follows:

```python
from .models import Marksheet, MarkType, ProjectAssignment, Project, StudentReport
from .forms import (
    MarksheetListForm,
    ProjectAssignmentForm,
    FinalizeMarksForm,
    ProjectCopySourceForm,
    ProjectAssignmentFilter,
    BulkMarksheetsImportForm,
    BulkTurnitinImportForm,
    BulkUploadMinervaAssignmentForm,
)
from .tables import BaseTable, MarkColumn, ProjectTable, AllProjectTable
```

In 5.1.2 nearly all of these symbols are correctly suppressed in the outline explorer EXCEPT for the import from .forms.

It's not as simple as the tuple syntax in the import as I also have the following lines which are all correctly suppressed:

```python
from phys_utils.views import (
    IsStaffViewMixin,
    IsSuperuserViewMixin,
    MultiFormMixin,
    FormListView,
)
from phys_utils.response import PDF_TemplateResponse, CSV_TemplateResponse
```

So it seems like it is a combination of both the relative package import and the tuple syntax. 

Unfortunately, black is rath fond of the tuple formatting of the long import :-) !

### LINKED ISSUE #16406: Outline not showing functions or erroneously showing imports as functions
### Issue Report Checklist

* [x ] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [ x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x ] Tried basic troubleshooting (if a bug/error)
    * [x ] Restarted Spyder
    * [x ] Reset preferences with ``spyder --reset``
    * [x ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x ] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

### What steps reproduce the problem?
1. outline not listing functions, doesn't matter if they are in code cells or not
image
2. outline listing module imports as generic functions when they are moved to a new line with "\"
image

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 5.1.3
* Python version: 3.7.11
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version:  windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.26.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.10.0 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.2 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.0 (OK)
qtpy >=1.5.0                  :  1.10.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  4.0.2 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.1 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.3 (OK)
zmq >=17                      :  22.2.1 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  3.4.2 (OK)
numpy >=1.7                   :  1.18.5 (OK)
pandas >=1.1.1                :  1.1.2 (OK)
scipy >=0.17.0                :  1.6.3 (OK)
sympy >=0.7.3                 :  None (OK)
```

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- Those are packages that don't have an `__init__.py` in the root of their repository.
- Also avoid showing relative imports in the Outline.
- Move `main_window` fixture to its corresponding `conftest.py` file to make it easier to contribute to it.
- Add tests to catch future regressions for these bugs.
- Depends on https://github.com/python-lsp/python-lsp-server/pull/261.

### Before

- No symbols for namespace packages

    image

- Showing relative imports in multiple lines as symbols

    image

### After

- Symbols are shown for namespace packages

    image

- Relative imports in multiple lines are not shown

    image

### Issue(s) Resolved

Fixes #16406.
Fixes #16352.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

File: spyder/app/tests/test_mainwindow.py [hunk 1]
```diff
@@ -72,13 +67,6 @@
...
-# Location of this file
-LOCATION = osp.realpath(osp.join(os.getcwd(), osp.dirname(__file__)))
-
-# Time to wait until the IPython console is ready to receive input
-# (in milliseconds)
-SHELL_TIMEOUT = 40000 if os.name == 'nt' else 20000
-
 # Need longer EVAL_TIMEOUT, because need to cythonize and C compile ".pyx" file
 # before import and eval it
 COMPILE_AND_EVAL_TIMEOUT = 30000
```
Comment (dalthviz) [inline spyder/app/tests/test_mainwindow.py:72]: Should this and other constants and utility functions below be moved also to the `conftest.py` file ?

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 ! Left a comment regarding the moving of other things from `test_mainwindow.py` to `conftest.py`.

Seems like a test on MacOS is failing (could it be caused due to the changes on the `python-lsp-server`?)

Also, could you add a screenshot of the outline before/after the changes here to understand better how the outline should be working? Thanks!

File: spyder/app/tests/test_mainwindow.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/app/tests/test_mainwindow.py:72]: Good idea. I'll push a commit moving everything that it's not a test to `conftest.py`.

**Comment by ccordoba12** [comment]
> Seems like a test on MacOS is failing (could it be caused due to the changes on the python-lsp-server?)

It seems more related to refactoring I did to move the `main_window` fixture to `conftest.py`. I'll take a look at it.

> Also, could you add a screenshot of the outline before/after the changes here to understand better how the outline should be working? Thanks!

Ok, will do.

**Review by dalthviz — COMMENTED** [review]
Thanks @ccordoba12 ! Checked locally and seems like this is working as expected and :+1:

I guess then this is only missing the merge of https://github.com/python-lsp/python-lsp-server/pull/261 and updating here the subrepo, right @ccordoba12 ?

**Comment by ccordoba12** [comment]
> I guess then this is only missing the merge of https://github.com/python-lsp/python-lsp-server/pull/261 and updating here the subrepo, right @ccordoba12 ?

Yep, I'll push with `ci skip` to not re-run our tests again because it's unnecessary.

**Review by dalthviz — APPROVED** [review]

**Comment by ccordoba12** [comment]
> could it be that other changes done for v1.6.0 to the python-language-server are causing completion tests to fail?

Yep, the error was caused by a new option added in PR https://github.com/python-lsp/python-lsp-server/pull/246, but I deactivated it here because it leads to repeated entries in our completion widget:

image

**Pull request merged**